### ISSUE TITLE
Publishing to private repo gets stuck

### ISSUE THREAD

**Issue by reflog** [pikapkg/pack#2]
Hi! I tried replacing microbundle with pika, added the pipelines, the build goes just fine, but publishing with:
`pack publish --registry http://myrepo`
shows:
```
? Select semver increment or specify new version patch  4.4.3
? Will bump from 4.4.2 to 4.4.3. Continue? Yes

[1/6] Prerequisite checks...

```
And doesn't proceed.
Any hints on how to debug this?

Also, how should publish be used in a CI scenario? Perform 'pack publish --no-publish' locally to bump the version and then do a full publish on the CI server?

Thanks for a very promising tool!

**Comment by FredKSchott** [comment]
Oh that's interesting. I haven't tested with a private registry, so I wouldn't be surprised if there are issues lurking there.

I quickly looked through the code, do you think it's failing here?

``` js
// src/util/publish/prerequisite.ts
export default async function prerequisites(input, pkg, options) {
  const isExternalRegistry = typeof pkg.publishConfig === 'object' && typeof pkg.publishConfig.registry === 'string';
  if (!(pkg.private || isExternalRegistry)) {
    await pTimeout(
      (async () => {
        try {
          await execa('npm', ['ping']);
        } catch (_) {
          throw new Error('Connection to npm registry failed');
        }
      })(),
      15000,
      'Connection to npm registry timed out',
    );
  }
```

It looks like we're expecting `publishConfig.registry` in the `package.json`, try adding the registry URL there and seeing if that fixes this.

**Comment by reflog** [comment]
Hi!
I've tried this approach, added:
```
 "publishConfig": {
    "access": "restricted",
    "registry": "http://....."
  },
```
but the result is the same :(

Here's my whole package.json:
```
{
  "name": "@mycompany/my-package",
  "version": "4.4.2",
  "description": "",
  "private": true,
  "@pika/pack": {
    "pipeline": [
      [
        "@pika/plugin-ts-standard-pkg"
      ],
      [
        "@pika/plugin-build-node"
      ],
      [
        "@pika/plugin-build-web"
      ],
      [
        "@pika/plugin-build-types"
      ]
    ]
  },
  "scripts": {
    "build": "pack build",
    "deploy": "pack publish"
  },
  "author": "",
  "license": "ISC",
  "publishConfig": {
    "access": "restricted",
    "registry": "http://.........."
  },
  "devDependencies": {
    "@pika/pack": "^0.3.1",
    "@pika/plugin-build-node": "0.3.9",
    "@pika/plugin-build-types": "0.3.9",
    "@pika/plugin-build-web": "0.3.9",
    "@pika/plugin-ts-standard-pkg": "0.3.9",
    "tslint": "^5.11.0",
    "typescript": "3.3.1"
  }
}
```

**Comment by FredKSchott** [comment]
Hmm, I just ran `pack publish --registry http://myrepo` on my machine to test and it completed the publish. Have you tried running `publish` on the public registry without this option, to confirm that it's the option itself?

**Comment by reflog** [comment]
You are right. It's not the registry.
Tried without it (removed publishConfig from package.json), same result:
```
$ C:\code\node_modules\.bin\pack publish
@pika/pack publish v0.3.1

Publish a new version of @riggoh/revenova-shared (current: 4.4.2)

Commits:
- .
- trying picka

Commit Range:
v4.4.1...master (undefined/compare/v4.4.1...master)

? Select semver increment or specify new version patch  4.4.3
? Will bump from 4.4.2 to 4.4.3. Continue? Yes

[1/6] Prerequisite checks...
```

Any way to enable more verbose logging on what's going on?

**Comment by FredKSchott** [comment]
Hey @reflog did you ever figure out this issue? `--verbose` is a supported option but I don't think publish was built to output anything extra with it (we're still in the middle of integrating the original code from np)

**Comment by stramel** [comment]
Let me know if I need to test it on my private registry as well

**Comment by reflog** [comment]
Hey guys. No didn't manage to find out the culprit. Seems to be some issue with BaseReporter and Windows not liking each other. But I cannot prove it properly :)

**Comment by JoshRosenstein** [comment]
@FredKSchott 
Having same issue with my scoped package. But after tinkering, i believe its a windows and execa issue.
Fiddled around with execa and whoami, and always got an error

I got it to publish by commenting out that section and hard entering my username, but for the future may be necessary to find an alternative way to check access.

Looking into some other packages-for instance yarn does a [simple access check ](https://github.com/yarnpkg/yarn/blob/697a254ee794085c77f82f14f4a515d05e81e294/src/cli/commands/publish.js#L36) and reports errors after from the PUT responses. 

lerna does something similar [here ](https://github.com/lerna/lerna/blob/86017d769bb83dfd1abb88f4cf560f73651ad494/commands/publish/lib/verify-npm-package-access.js#L18) but uses [libnpmaccess](https://github.com/npm/libnpmaccess)

**Comment by JoshRosenstein** [comment]
Also for reference: just found [NPM need auth auth required for publishing` error (`npm whoami` shows logged in as correct user)](https://github.com/lerna/lerna/issues/896), lerna was having similar issues and the [Use npmClient instead of hardcoded npm](https://github.com/lerna/lerna/pull/1145) PR, fixed it.

**Comment by FredKSchott** [comment]
Should be fixed by #38, going out in a patch in a little bit.
